reporter/base_reporter.go

```diff
@@ -96,12 +96,14 @@ func (b *baseReporter) ReportTraceEvent(trace *libpf.Trace, meta *samples.TraceE
 		extraMeta = b.cfg.ExtraSampleAttrProd.CollectExtraSampleMeta(trace, meta)
 	}
 
-	containerID, err := libpf.LookupCgroupv2(b.cgroupv2ID, meta.PID)
+	cgroupv2Path, err := libpf.LookupCgroupv2(b.cgroupv2ID, meta.PID)
 	if err != nil {
 		log.Debugf("Failed to get a cgroupv2 ID as container ID for PID %d: %v",
 			meta.PID, err)
 	}
 
+	containerID := extractContainerID(cgroupv2Path)
```

Comment:
This does not cache the extracted container ID and therefore performing regex search for the same cgroup multiple times which should be avoided.

Reply:
As https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/535 was not well received, an extra cache is used. If the request is, to just use one cache, then I will end up with https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/535 again.

Reply:
I guess, with the latest https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/548/commits/aa21cdb7705c55d5f2695977493281c8f44e719a, there is the full circle back to https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/535.